### PR TITLE
8276186: Require getAvailableLocales() methods to include Locale.ROOT

### PR DESCRIPTION
**Pull request by naotoj**
This fix is to require to include `Locale.ROOT` in the returned arrays/set from `getAvailableLocales()` methods in various locale-sensitive classes. The implementation has been including `Locale.ROOT` since its inception, it is simply a doc clarification (+ a test case verifying it). Corresponding CSR has also been drafted: https://bugs.openjdk.java.net/browse/JDK-8276249

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8276186](https://bugs.openjdk.java.net/browse/JDK-8276186): Require getAvailableLocales() methods to include Locale.ROOT

### Reviewers
 * [Pavel Rappo](https://openjdk.java.net/census#prappo) (@pavelrappo - **Reviewer**) ⚠️ Review applies to 1a611bd50a63647d35f50e8cd41d1e58eaaa70c0
 * [Stuart Marks](https://openjdk.java.net/census#smarks) (@stuart-marks - **Reviewer**)
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6258/head:pull/6258` \
`$ git checkout pull/6258`

Update a local copy of the PR: \
`$ git checkout pull/6258` \
`$ git pull https://git.openjdk.java.net/jdk pull/6258/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6258`

View PR using the GUI difftool: \
`$ git pr show -t 6258`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6258.diff">https://git.openjdk.java.net/jdk/pull/6258.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/time/format/DecimalStyle.java [hunk 1]
```diff
@@ -115,7 +115,9 @@
     /**
      * Lists all the locales that are supported.
      * <p>
-     * The locale 'en_US' will always be present.
+     * At a minimum, the returned Set must contain a {@code Locale} instance equal to
```
Comment (pavelrappo) [inline src/java.base/share/classes/java/time/format/DecimalStyle.java:118]: A nit, really. Consider applying either of these suggestions:
```suggestion
     * At a minimum, the returned {@code Set} must contain a {@code Locale} instance equal to
```
```suggestion
     * At a minimum, the returned set must contain a {@code Locale} instance equal to
```

**Review by pavelrappo — APPROVED** [review]
The change to the existing source looks good, and the new test looks beautiful. Thanks for doing this.

File: src/java.base/share/classes/java/time/format/DecimalStyle.java (hunk 1 above)
Comment (naotoj) [reply src/java.base/share/classes/java/time/format/DecimalStyle.java]: Thanks. Adopted the former suggestion.

**Review by stuart-marks — APPROVED** [review]
Nice use of MethodHandles.lookup() in the test.

**Review by irisclark — APPROVED** [review]

**Pull request closed without merging**